tests/setup_test.go

```diff
@@ -3130,3 +3134,442 @@ func TestGetUsers(t *testing.T) {
+
+	_ = testutil.Wait(30*time.Second, func() (bool, error) {
+		runs, _, err := gwClient.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+		if err != nil {
+			return false, nil
+		}
+
+		if len(runs) == 0 {
+			return false, nil
+		}
+		run := runs[0]
+		if run.Phase != rstypes.RunPhaseFinished {
+			return false, nil
+		}
+
+		return true, nil
+	})
+
+	runs, _, err := gwClient.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	if len(runs) != 1 {
+		t.Fatalf("expected %d run got: %d", 1, len(runs))
+	}
+
+	gwClient = gwclient.NewClient(sc.config.Gateway.APIExposedURL, sc.config.Gateway.AdminToken)
+
+	users, _, err := gwClient.GetUsers(ctx, "", 0, false)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	projectgroup, _, err := gwClient.GetProjectGroup(ctx, "user/user01")
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	remotesources, _, err := gwClient.GetRemoteSources(ctx, "", 0, false)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	user01Projects, _, err := gwClient.GetProjectGroupProjects(ctx, "user/user01")
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	w, err := os.Create(filepath.Join(dir, "export-configstore"))
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	resp, err := gwClient.Export(ctx, configstoreService)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	_, err = io.Copy(w, resp.Body)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	w, err = os.Create(filepath.Join(dir, "export-runservice"))
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	resp, err = gwClient.Export(ctx, runserviceService)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	_, err = io.Copy(w, resp.Body)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	_, err = gwClient.EnableMaintenance(ctx, configstoreService)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	_, err = gwClient.EnableMaintenance(ctx, runserviceService)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	time.Sleep(3 * time.Second)
+
+	r, err := os.Open(filepath.Join(dir, "export-configstore"))
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	_, err = gwClient.Import(ctx, configstoreService, r)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	r, err = os.Open(filepath.Join(dir, "export-runservice"))
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	_, err = gwClient.Import(ctx, runserviceService, r)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	_, err = gwClient.DisableMaintenance(ctx, configstoreService)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	_, err = gwClient.DisableMaintenance(ctx, runserviceService)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	time.Sleep(3 * time.Second)
```

Comment:
There should be a way to check if maintenance mode is started instead of using a sleep.